scripts/make-bundle

```diff
@@ -12,171 +12,196 @@ interface for those not familiar with the python
-    awscli_dir = os.path.dirname(
-        os.path.dirname(os.path.abspath(__file__)))
-    with cd(scratch_dir):
-        run('%s install %s -d . -e %s' % (pip, INSTALL_ARGS, awscli_dir))
-    # Remove the awscli package, we'll create our own sdist.
-    _remove_cli_zip(scratch_dir)
-
-
-def _remove_cli_zip(scratch_dir):
-    clidir = [f for f in os.listdir(scratch_dir) if f.startswith('awscli')]
-    assert len(clidir) == 1
-    os.remove(os.path.join(scratch_dir, clidir[0]))
-
-
-def add_cli_sdist(scratch_dir):
-    awscli_dir = os.path.dirname(
-        os.path.dirname(os.path.abspath(__file__)))
-    if os.path.exists(os.path.join(awscli_dir, 'dist')):
-        shutil.rmtree(os.path.join(awscli_dir, 'dist'))
-    with cd(awscli_dir):
-        run('python setup.py sdist')
-        filename = os.listdir('dist')[0]
-        shutil.move(os.path.join('dist', filename),
-                    os.path.join(scratch_dir, filename))
-
-
-def create_bootstrap_script(scratch_dir):
+
+
+PYTHON_VERSIONS = ["2.6", "2.7", "3.3", "3.4", "3.5", "3.6"]
+
+
+def pip(python_version):
+    if python_version == "2.6":
+        return "pip.__main__"
+    return "pip"
+
+
+def ensure_packages(requirements, *, output_dir, pythons=None):
+    if pythons is None:
+        pythons = PYTHON_VERSIONS
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # We want to loop over every version of Python we support and download
+        # or build wheels for them.
+        first = False
+        for python in sorted(pythons, reverse=True):
+            print("Downloading Dependencies for {}".format(python))
+
+            # Create our virtual environment.
+            venv_path = os.path.abspath(os.path.join(tmpdir, python))
+            python_path = os.path.join(venv_path, "bin", "python")
+            subprocess.run(
+                ["virtualenv", "-ppython{}".format(python), venv_path],
+                stdout=subprocess.PIPE,
+                check=True,
+            )
+
+            # Ensure that we have the latest versions of all of our tools
+            # installed in the virtual environment.
+            subprocess.run(
+                [python_path, "-m", pip(python), "install", "--upgrade",
+                 "pip"],
+                stdout=subprocess.PIPE,
+                check=True,
+            )
+            subprocess.run(
+                [python_path, "-m", pip(python), "install", "--upgrade",
+                 "setuptools", "wheel"],
+                stdout=subprocess.PIPE,
+                check=True,
+            )
+
+            if not first:
+                first = True
+                # We need to download and bundle virtualenv, but we need it to
+                # be the sdist since it won't actually be installed.
+                subprocess.run(
+                    [python_path, "-m", pip(python), "download",
+                     "--no-cache-dir", "--no-binary=virtualenv",
+                     "-d", output_dir, "virtualenv"],
```

Comment:
Yeah I think I would feel more comfortable if we did. If there is a new functionality that we want to pick up, bumping the version and testing out the new installers have not been difficult. There really has not been any issues in the past, but it is better to be safe.